python/tests/datasources/test_census_pop_estimates_sc.py

```diff
@@ -0,0 +1,134 @@
+from unittest import mock
+import os
+import pandas as pd
+from pandas._testing import assert_frame_equal
+
+from datasources.census_pop_estimates_sc import CensusPopEstimatesSC, generate_pop_data_18plus
+
+# Current working directory.
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+TEST_DIR = os.path.join(THIS_DIR, os.pardir, "data", "census_pop_estimates_sc")
+
+STATE_POP_RACE_DATA = os.path.join(
+    TEST_DIR, 'census_pop_estimates_sc-race_ethnicity_age_state.csv')
+NATIONAL_POP_RACE_DATA = os.path.join(
+    TEST_DIR, 'census_pop_estimates_sc-race_ethnicity_age_national.csv')
+
+STATE_POP_SEX_DATA = os.path.join(
+    TEST_DIR, 'census_pop_estimates_sc-sex_age_state.csv')
+NATIONAL_POP_SEX_DATA = os.path.join(
+    TEST_DIR, 'census_pop_estimates_sc-sex_age_national.csv')
+
+
+def get_pop_estimates_as_df():
+    print("MOCK FILE READ OF sc-est2021-alldata6.csv")
+    return pd.read_csv(os.path.join(TEST_DIR, 'sc-est2021-alldata6.csv'), dtype={
+        'STATE': str,
+        'STNAME': str,
+    })
+
+
+def get_breakdown_df():
+    return pd.DataFrame({
+        "col1": [0, 1, 2],
+        "col2": ["a", "b", "c"]
+    })
+
+
+# TEST OVERALL WRITE TO BQ
+
+@mock.patch('datasources.census_pop_estimates_sc.generate_pop_data_18plus',
+            return_value=get_breakdown_df())
+@mock.patch('ingestion.gcs_to_bq_util.load_csv_as_df_from_web',
+            return_value=get_pop_estimates_as_df())
+@mock.patch('ingestion.gcs_to_bq_util.add_df_to_bq',
+            return_value=None)
+def testWriteToBq(
+    mock_bq: mock.MagicMock,
+    mock_csv: mock.MagicMock,
+    mock_gen: mock.MagicMock
+):
+
+    censusPopEstimatesSC = CensusPopEstimatesSC()
+
+    kwargs = {'filename': 'test_file.csv',
+              'metadata_table_id': 'test_metadata',
+              'table_name': 'output_table'}
+
+    censusPopEstimatesSC.write_to_bq('dataset', 'gcs_bucket', **kwargs)
+    assert mock_csv.call_count == 1
+
+    # 4 = 2 demographic breakdowns X 2 geographic breakdowns
+    assert mock_bq.call_count == 4
+    assert mock_gen.call_count == 4
+
+# TEST INNER FUNCTION - RACE BY STATE
+
+
+def test18PlusByRace():
```

Comment:
Can you specify these are by state in the function title.

Reply:
good call